### ISSUE TITLE
[cmake] LAPACK detection is broken on Ubuntu 16.04 with system tools (Trac #2103)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2095]
It can find the library through pkg-config, but doesn't set LAPACK_FOUND.

-- blas and lapack 
-- Searching for OpenBLAS in IceCube's CVMFS (or Arch's AUR)
CMake Warning at cmake/tools/blas.cmake:59 (find_package):
  By not providing "FindOpenBLAS.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "OpenBLAS",
  but CMake did not find one.

  Could not find a package configuration file provided by "OpenBLAS" with any
  of the following names:

    OpenBLASConfig.cmake
    openblas-config.cmake

  Add the installation prefix of "OpenBLAS" to CMAKE_PREFIX_PATH or set
  "OpenBLAS_DIR" to a directory containing one of the above files.  If
  "OpenBLAS" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  cmake/tools.cmake:60 (include)
  CMakeLists.txt:99 (include)

-- Searching for OpenBLAS via pkg-config
-- + OpenBLAS
-- +  version: 0.2.18
-- + includes: /usr/include/openblas
-- +     libs: /usr/lib/libopenblas.so
-- +   lapack: /usr/lib/libopenblas.so;/usr/lib/openblas-base/liblapack.so

...

-- + millipede
-- +-- python [symlinks] 
CMake Error at cmake/tools.cmake:71 (message):
  Attempt to use tool 'lapack' which wasn't found

Manually setting LAPACK_FOUND in blas.cmake fixes this, but this seems hacky.  I thought find_library set this.

Migrated from https://code.icecube.wisc.edu/ticket/2103
```json
{
    "status": "closed", 
    "changetime": "2019-01-12T00:34:05", 
    "description": "It can find the library through pkg-config, but doesn't set LAPACK_FOUND.\n\n\n-- blas and lapack \n-- Searching for OpenBLAS in IceCube's CVMFS (or Arch's AUR)\nCMake Warning at cmake/tools/blas.cmake:59 (find_package):\n  By not providing \"FindOpenBLAS.cmake\" in CMAKE_MODULE_PATH this project has\n  asked CMake to find a package configuration file provided by \"OpenBLAS\",\n  but CMake did not find one.\n\n  Could not find a package configuration file provided by \"OpenBLAS\" with any\n  of the following names:\n\n    OpenBLASConfig.cmake\n    openblas-config.cmake\n\n  Add the installation prefix of \"OpenBLAS\" to CMAKE_PREFIX_PATH or set\n  \"OpenBLAS_DIR\" to a directory containing one of the above files.  If\n  \"OpenBLAS\" provides a separate development package or SDK, be sure it has\n  been installed.\nCall Stack (most recent call first):\n  cmake/tools.cmake:60 (include)\n  CMakeLists.txt:99 (include)\n\n\n-- Searching for OpenBLAS via pkg-config\n-- + OpenBLAS\n-- +  version: 0.2.18\n-- + includes: /usr/include/openblas\n-- +     libs: /usr/lib/libopenblas.so\n-- +   lapack: /usr/lib/libopenblas.so;/usr/lib/openblas-base/liblapack.so\n\n...\n\n\n-- + millipede\n-- +-- python [symlinks] \nCMake Error at cmake/tools.cmake:71 (message):\n  Attempt to use tool 'lapack' which wasn't found\n\n\nManually setting LAPACK_FOUND in blas.cmake fixes this, but this seems hacky.  I thought find_library set this.", 
    "reporter": "olivas", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547253245781054", 
    "component": "cmake", 
    "summary": "[cmake] LAPACK detection is broken on Ubuntu 16.04 with system tools", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2017-10-18T01:26:15", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20171018T01:33:20: david.schultz commented:

> No, `find_library()` only sets the variable name you give it, which in this case is `LAPACK_LIBRARIES`.  `find_package()` is what you are thinking of that does this.

**Comment by icecube-trac** [comment]
Trac update at 20171018T01:42:26: olivas commented:

> http://code.icecube.wisc.edu/projects/icecube/changeset/2989/IceTray
> 
> That fixes it on my machine, but I'm going to leave this open so Don can double-check that.
>

**Comment by icecube-trac** [comment]
Trac update at 20171018T20:00:48: nega commented:

> No no, that's correct. Cut-n-paste error on my part I guess.
> 
> I'm going to leave this open a bit. Looking at it today, I don't like the OpenBLAS case as much.

**Comment by icecube-trac** [comment]
Trac update at 20171018T20:53:33: 
* nega commented:

> In changeset 2997:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2997"
> rejigger the "ubuntu openblas" detection block
> 
> additionally make the "specific atlas" detection block look like the above.
> 
> Successfuly OpenBLAS detection on Ubuntu 16.04 now looks like:
> 
>  blas and lapack 
>  Searching for OpenBLAS in IceCube's CVMFS (or Arch's AUR)
>  Searching for OpenBLAS via pkg-config
>  + OpenBLAS
>  +  version: 0.2.18
>  + includes: /usr/include/openblas
>  +     libs: /usr/lib/libopenblas.so
>  +   lapack: /usr/lib/libopenblas.so;/usr/lib/openblas-base/liblapack.so
> 
> closes #2095
> ```


* nega changed resolution from "" to "fixed"
* nega changed status from "new" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:34:05: nwhitehorn commented:

> In changeset 170412:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170412"
> rejigger the "ubuntu openblas" detection block
> 
> additionally make the "specific atlas" detection block look like the above.
> 
> Successfuly OpenBLAS detection on Ubuntu 16.04 now looks like:
> 
>  blas and lapack 
>  Searching for OpenBLAS in IceCube's CVMFS (or Arch's AUR)
>  Searching for OpenBLAS via pkg-config
>  + OpenBLAS
>  +  version: 0.2.18
>  + includes: /usr/include/openblas
>  +     libs: /usr/lib/libopenblas.so
>  +   lapack: /usr/lib/libopenblas.so;/usr/lib/openblas-base/liblapack.so
> 
> closes #2095
> ```
